### ISSUE TITLE
License needs clarity on main README

### ISSUE THREAD

**Issue by fujifish** [Unitech/pm2#1196]
Currently the main README states that PM2 is licensed under AGPLv3. However, the ADVANCED_README states that in addition to AGPLv3, CLI.js is licensed under Apache license.
The main README should reflect the same as the ADVANCED_README.

Also, it would be great if the main README (for clarity) would state that: 
"It is allowed to redistribute PM2 as-is and also to embed CLI.js in your code to control PM2 without turning the embedding code into AGPLv3"

**Comment by RobertoMalatesta** [comment]
+1 : This a SEVERE ISSUE for any possible enterprise use.
Can you please move to an Apache License or some more MIT-like license?

**Comment by markrey** [comment]
+1 It would be great to go with one license.
Apache , MIT would be great. AGPL is big NO for most of the enterprises.

**Comment by harking** [comment]
+1 for MIT or Apache license.

**Comment by darthcav** [comment]
We also share this concern in regard to the licensing of the framework. We will welcome more clarity on this issue.

**Comment by jvecchioli** [comment]
Same concern here. 
Can something be done on this licensing issue?

**Comment by hirako2000** [comment]
In the community dominated by the MIT/Apache software, it is really suspicious to license pm2 as GPL.
The contagion just makes it unbearable and actually prevents a lot of MIT libraries to pull pm2 dependency to provide pm2 out of the box, at risk to have to move to GPL as well.

What's the reasoning behind this?

**Comment by FOSS117** [comment]
Given https://github.com/Unitech/pm2/blob/master/README.md#license -- would maintainers simply consider switching to an LGPL 2.1 license for pm2 API's use only ? Current AGPL terms definitely causes heartburn for enterprise use.

**Comment by ryan-lane** [comment]
I'm pretty iffy on the licensing here in general. CLI is importing stuff from lib, so doesn't that make it AGPL?

It's really common for people to import pm2 in their code to make a more custom launcher, which is going to pollute their code without them knowing. pm2 is effectively a library in a lot of cases, and AGPL is a really aggressive choice for a library that's mostly going to screw over people who aren't paying attention, or don't know any better. It's really likely that people are pulling this in transitively and have license violations without knowing.

The sad thing is it doesn't look like there's a CLA or any type of copyright assignment for this project, so I'm guessing it's actually impossible to re-license at this point :(.

**Comment by maziyarpanahi** [comment]
The part that says `For other licenses contact us.` clarifies this to me pretty much! If you feel iffy about the licensing, it means you need to contact them and pay for the appropriate license.
Even though I am an indie developer whom nobody cares what I make and how I make it let along to come after me, I still feel like my old forever.js is at least clear as for how I could use it.

**Comment by samsch** [comment]
This is still an unresolved issue.

**Comment by harking** [comment]
This is still unresolved

**Comment by shayneoneill** [comment]
To be honest, even regular GPL would be an improvement. I don't think most use cases are actually legal with the AGPL.

**Comment by Calabacin** [comment]
It is unbelievable that this is still not addressed or cleared up after so long. I think that the license should not be a problem, but lack of clarifications makes me cautious.

We stopped using PM2 about 5 years ago due to this doubt not being addressed. We had to migrate several projects and it was costly. Today someone asked if they could use pm2 in another project; seeing this the answer will again be NO.
